update-controlled-shapes.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import typer
+import requests
+from requests.exceptions import HTTPError
+import json
+from rdflib.namespace import RDF, RDFS, VOID, XSD, SOSA, DCTERMS, PROV, TIME, SH
+from rdflib import Graph, Namespace, URIRef, Literal, BNode
+from rdflib.collection import Collection
+
+app = typer.Typer()
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class SPARQLQueryError(Exception):
+    pass
+
+
+def sparql_query(url: str, query: str):
+    headers = {
+        "Content-type": "application/sparql-query",
+        "Accept": "application/sparql-results+json",
+    }
+    r = requests.post(url, headers=headers, data=query)
+    try:
+        r.raise_for_status()
+    except HTTPError as err:
+        raise SPARQLQueryError(
+            f"Failed fetching data from {url} SPARQL endpoint. Code: {r.status_code} Message: {r.text}."
+        ) from err
+    return r.json()
+
+
+@app.command()
+def fetch(filename: str):
+    errors = 0
+
+    g = Graph()
+
+    g.parse(filename)
+
+    q = """
+        select ?uri ?query ?endpoint {
+            ?uri a <urn:class:Controlled> ;
+            <urn:property:query> ?query ;
+            <urn:property:sparqlEndpoint> ?endpoint .
+
+        }
+    """
+
+    for shape in g.query(q).bindings:
+
+        start_from_the_last_blank_node = True
+        uri = shape["uri"]
+        query = shape["query"]
+        endpoint = shape["endpoint"]
+
+        try:
+            data = sparql_query(endpoint, query)
+
+            sh_in_value = g.value(uri, SH["in"])
+            old_list = Collection(g, sh_in_value)
+            old_list.clear()
+            g.remove((uri, SH["in"], None))
+
+            for r in data["results"]["bindings"]:
+                if start_from_the_last_blank_node:
+                    last_node = BNode()
+                    g.add((last_node, RDF.first, URIRef(r["values"]["value"])))
+                    g.add((last_node, RDF.rest, RDF.nil))
+                    start_from_the_last_blank_node = False
+                else:
+                    next_node = BNode()
+                    g.add((next_node, RDF.first, URIRef(r["values"]["value"])))
+                    g.add((next_node, RDF.rest, last_node))
+                    last_node = next_node
+
+            g.add((uri, SH["in"], last_node))
```

Comment:
No need to remove `uri sh:in`. 

The instance of `Collection` can be used to clear the list as well as add to the list just like a normal Python list.

E.g.,

```python
collection = Collection(g, sh_in_value)
collection.clear()

for item in results:
    collection.append(item)
```